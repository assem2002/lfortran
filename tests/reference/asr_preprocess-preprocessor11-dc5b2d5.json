{
    "basename": "asr_preprocess-preprocessor11-dc5b2d5",
    "cmd": "lfortran --cpp --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/preprocessor11.f90",
    "infile_hash": "dd7b363224d526463b7ce535007b839ea7d9264d1b28f7bf7bbf6e7b",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr_preprocess-preprocessor11-dc5b2d5.stdout",
    "stdout_hash": "a8b2046e1d2cb8576f63ca9593b5e284f91ae632abe2f44eb9a063a5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}