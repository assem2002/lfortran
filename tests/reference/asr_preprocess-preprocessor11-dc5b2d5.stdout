(TranslationUnit
    (SymbolTable
        1
        {
            preprocessor11:
                (Program
                    (SymbolTable
                        2
                        {
                            
                        })
                    preprocessor11
                    []
                    [(Print
                        [(StringFormat
                            ()
                            [(IntegerBinOp
                                (IntegerConstant 1 (Integer 4) Decimal)
                                Add
                                (IntegerBinOp
                                    (IntegerConstant 5 (Integer 4) Decimal)
                                    Mul
                                    (IntegerConstant 3 (Integer 4) Decimal)
                                    (Integer 4)
                                    (IntegerConstant 15 (Integer 4) Decimal)
                                )
                                (Integer 4)
                                (IntegerConstant 16 (Integer 4) Decimal)
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "X is 1"
                                (Character 1 6 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "X is not 2"
                                (Character 1 10 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Not X is not 1"
                                (Character 1 14 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "X is 1 and Y is 5"
                                (Character 1 17 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Not (X is 2 and Y is 5)"
                                (Character 1 23 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "X<3 and Y>3"
                                (Character 1 11 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Not (X>3 and Y<3)"
                                (Character 1 17 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "X is 3-2"
                                (Character 1 8 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "X is not 3-1"
                                (Character 1 12 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y is (6*5-5)/5"
                                (Character 1 14 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y is not (6*5-6)/5"
                                (Character 1 18 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y is (-5+6*5)/5"
                                (Character 1 15 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y is not (-6+6*5)/5"
                                (Character 1 19 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y is (+5+6*5)/7"
                                (Character 1 15 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y is not (+4+6*5)/7"
                                (Character 1 19 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y is (+5+6*Y)/7"
                                (Character 1 15 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Z is 16"
                                (Character 1 7 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Y defined and 1 <= Y"
                                (Character 1 20 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "A not defined or not 1 <= A"
                                (Character 1 27 ())
                            )]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )]
                )
        })
    []
)
